src/App.tsx

```diff
@@ -1,24 +1,192 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
+import { Error } from './types/Error';
+import { TodoCondition } from './types/TodoCondition';
 
-const USER_ID = 0;
+import { filterTodos } from './utils/filterTodos';
+import {
+  USER_ID,
+  deleteTodo,
+  getTodos,
+  changeTodo,
+} from './api/todos';
+
+import { UserWarning } from './UserWarning';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { TodoItem } from './components/TodoItem';
+import { Footer } from './components/Footer/Footer';
+import { ErrorMessage } from './components/ErrorMessage';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<Filter>(Filter.All);
+  const [errorType, setErrorType] = useState(Error.None);
+  const [todoCondition, setTodoCondition]
+    = useState<TodoCondition>(TodoCondition.neutral);
+  const [procesingTodosId, setProcesingTodosId] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const handleError = (err: Error) => {
+    if (err !== Error.None) {
+      setTimeout(() => setErrorType(Error.None), 3000);
+    }
+
+    setErrorType(err);
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(result => {
+        setTodos(result);
+      })
+      .catch(() => handleError(Error.Load));
+  }, []);
+
+  const todosStatus = useMemo(() => {
+    return {
+      isActive: todos.some(todo => todo.completed === false),
+      isCompleted: todos.some(todo => todo.completed === true),
+    };
+  }, [todos]);
+
+  const todoDelete = (todoId: number) => {
```

Comment:
```suggestion
  const deleteTodo = (todoId: number) => {
```
general convetntion that first word in function name should be a verb